### PR TITLE
feat(dop,pipeline): Decoupling gittar from pipeline

### PR DESCRIPTION
**Pull request by littlejiancc**
#### What this PR does / why we need it:
Decoupling gittar from pipeline

#### Which issue(s) this PR fixes:

- Fixes #your-issue_number
- [Erda Cloud Issue Link](https://erda.cloud/erda/dop/projects/387/issues/gantt?filter__urlQuery=eyJpdGVyYXRpb25JRHMiOlsxMTc4LDExNjRdLCJhc3NpZ25lZSI6WyIxMDAxMjYxIl19&id=299845&iterationID=1164&pId=267098&type=TASK)

#### Specified Reviewers:

/assign @sfwn 

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |     Decoupling gittar from pipeline         |
| 🇨🇳 中文    |      pipeline解耦gittar        |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: api/proto-go/core/pipeline/base/pb/base.pb.go [hunk 1]
```diff
@@ -4359,6 +4359,10 @@ type PipelineCreateRequest struct {
 	// +optional
 	DefinitionID string           `protobuf:"bytes,18,opt,name=definitionID,proto3" json:"definitionID,omitempty"`
 	IdentityInfo *pb.IdentityInfo `protobuf:"bytes,19,opt,name=identityInfo,json=-,proto3" json:"identityInfo,omitempty"`
+	// Secrets Environment variables passed from the upper layer，different from configs cms
```
Comment (sfwn) [inline api/proto-go/core/pipeline/base/pb/base.pb.go:4362]: Secrets variables, not env.

Comment (sfwn) [reply api/proto-go/core/pipeline/base/pb/base.pb.go:4362]: passed from the invoker, different from config cms

Comment (sfwn) [reply api/proto-go/core/pipeline/base/pb/base.pb.go:4362]: passed from the invoker, different from config cms

File: api/proto-go/all.go [hunk 2]
```diff
@@ -1,17 +1,3 @@
-// Copyright (c) 2021 Terminus, Inc.
```
Comment (sfwn) [inline api/proto-go/all.go:1]: do not commit these changes. too many irrelevant files.
or commit in another pr.

Comment (sfwn) [inline api/proto-go/all.go:1]: do not commit these changes. too many irrelevant files.
or commit in another pr.

File: modules/dop/utils/gittar.go [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+	"github.com/erda-project/erda/modules/dop/conf"
+	"github.com/erda-project/erda/pkg/discover"
+	"github.com/erda-project/erda/pkg/http/httpclientutil"
+)
+
+func GetGittarRepoURL(clusterName, repoAbbr string) string {
+	return getCenterOrSaaSURL(conf.DiceClusterName(), clusterName, httpclientutil.WrapHttp(discover.Gittar()), httpclientutil.WrapHttp(conf.GittarPublicURL())) + "/" + repoAbbr
+}
+
+func getCenterOrSaaSURL(diceCluster, requestCluster, center, sass string) string {
```
Comment (sfwn) [inline modules/dop/utils/gittar.go:28]: getCenterOrEdgeURL

Comment (sfwn) [inline modules/dop/utils/gittar.go:28]: getCenterOrEdgeURL

File: modules/pipeline/spec/pipeline_extra.go [hunk 4]
```diff
@@ -70,8 +70,9 @@ type PipelineExtraInfo struct {
 	// Deprecated
 	ConfigManageNamespaceOfSecretsDefault string `json:"configManageNamespaceOfSecretsDefault,omitempty"`
 	// Deprecated
-	ConfigManageNamespaceOfSecrets string   `json:"configManageNamespaceOfSecrets,omitempty"`
-	ConfigManageNamespaces         []string `json:"configManageNamespaces,omitempty"`
+	ConfigManageNamespaceOfSecrets string            `json:"configManageNamespaceOfSecrets,omitempty"`
+	ConfigManageNamespaces         []string          `json:"configManageNamespaces,omitempty"`
+	Secrets                        map[string]string `json:"secrets,omitempty"`
```
Comment (sfwn) [inline modules/pipeline/spec/pipeline_extra.go:75]: IncomingSecrets

Comment (sfwn) [inline modules/pipeline/spec/pipeline_extra.go:75]: IncomingSecrets

File: modules/pipeline/spec/pipeline_cron.go [hunk 5]
```diff
@@ -60,6 +60,7 @@ type PipelineCronExtra struct {
 	NormalLabels           map[string]string `json:"normalLabels"` // userID 存储提交流水线的用户 ID
 	Envs                   map[string]string `json:"envs"`
 	ConfigManageNamespaces []string          `json:"configManageNamespaces,omitempty"`
+	Secrets                map[string]string `json:"secrets,omitempty"`
```
Comment (sfwn) [inline modules/pipeline/spec/pipeline_cron.go:63]: IncomingSecrets

Comment (sfwn) [inline modules/pipeline/spec/pipeline_cron.go:63]: IncomingSecrets

**Review by sfwn — CHANGES_REQUESTED** [review]

**Comment by sfwn** [comment]
Test in bigdata and cron scenario especially.

File: api/proto-go/all.go (hunk 2 above)
Comment (littlejiancc) [reply api/proto-go/all.go]: done

Comment (littlejiancc) [reply api/proto-go/all.go]: done

File: api/proto-go/core/pipeline/base/pb/base.pb.go (hunk 1 above)
Comment (littlejiancc) [reply api/proto-go/core/pipeline/base/pb/base.pb.go:4362]: done

File: modules/dop/utils/gittar.go (hunk 3 above)
Comment (littlejiancc) [reply modules/dop/utils/gittar.go:28]: done

Comment (littlejiancc) [reply modules/dop/utils/gittar.go:28]: done

File: modules/pipeline/spec/pipeline_extra.go (hunk 4 above)
Comment (littlejiancc) [reply modules/pipeline/spec/pipeline_extra.go:75]: done

Comment (littlejiancc) [reply modules/pipeline/spec/pipeline_extra.go:75]: done

File: modules/pipeline/services/pipelinesvc/batch_create.go [hunk 6]
```diff
@@ -54,7 +55,8 @@ func (s *PipelineSvc) BatchCreate(batchReq *apistructs.PipelineBatchCreateReques
 		if batchReq.AutoRun {
```
Comment (sfwn) [inline modules/pipeline/services/pipelinesvc/batch_create.go:55]: how about batch-create of cron but not auto run?

**Review by sfwn — CHANGES_REQUESTED** [review]

File: modules/pipeline/services/pipelinesvc/batch_create.go (hunk 6 above)
Comment (littlejiancc) [reply modules/pipeline/services/pipelinesvc/batch_create.go:55]: cron is also compatibled in cron_task.go

File: modules/pipeline/spec/pipeline_cron.go (hunk 5 above)
Comment (littlejiancc) [reply modules/pipeline/spec/pipeline_cron.go]: done

File: api/proto-go/core/pipeline/base/pb/base.pb.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (littlejiancc) [reply api/proto-go/core/pipeline/base/pb/base.pb.go]: done

**Review by sfwn — APPROVED** [review]

File: modules/pipeline/services/pipelinesvc/batch_create.go [hunk 7]
```diff
@@ -64,3 +66,20 @@ func (s *PipelineSvc) BatchCreate(batchReq *apistructs.PipelineBatchCreateReques
 
 	return result, nil
 }
+
+// getSecrets Compatible with bigdata application
+func getSecrets(p *spec.Pipeline) map[string]string {
```
Comment (sfwn) [inline modules/pipeline/services/pipelinesvc/batch_create.go:71]: add todo for this.

**Review by sfwn — CHANGES_REQUESTED** [review]

**Review by sfwn — APPROVED** [review]

**Pull request merged**